testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/management/cli/UndeployTestCase.java

```diff
@@ -92,57 +85,67 @@ public static void after() throws Exception {
 
     @After
     public void afterTest() throws Exception {
+        ctx.handleSafe("deployment undeploy * --all-relevant-server-groups");
+
         ctx.terminateSession();
+        infoUtils.resetDoubleCheck();
     }
 
+    /**
+     * Test undeploying operation by one application deployment to undeploy all by wildcard in all server group.
+     * Uses legacy commands.
+     *
+     * @throws Exception
+     */
     @Test
-    public void undeploy() throws Exception {
+    public void testLegacyUndeployWithAllServerGroups() throws Exception {
+        ctx.handle("deploy --server-groups=" + sgOne + ',' + sgTwo + " " + cliTestApp1War.getAbsolutePath());
+
+        infoUtils.checkDeploymentByLegacyInfo(sgOne, cliTestApp1War.getName(), ENABLED);
+        infoUtils.checkDeploymentByLegacyInfo(sgTwo, cliTestApp1War.getName(), ENABLED);
+
         // From serverGroup1 only, still referenced from sg2. Must keep-content
         ctx.handle("undeploy --server-groups=" + sgOne + ' ' + cliTestApp1War.getName());
 
-        checkState(sgOne, cliTestApp1War.getName(), false);
-        checkState(sgTwo, cliTestApp1War.getName(), true);
+        infoUtils.checkDeploymentByLegacyInfo(sgOne, cliTestApp1War.getName(), ADDED);
+        infoUtils.checkDeploymentByLegacyInfo(sgTwo, cliTestApp1War.getName(), ENABLED);
 
-        ctx.handle("deployment undeploy * --all-relevant-server-groups");
+        ctx.handle("undeploy --name=* --server-groups=" + sgOne + ',' + sgTwo);
+
+        infoUtils.readLegacyDeploymentInfo(sgOne);
+        infoUtils.checkMissingInOutputMemory(cliTestApp1War.getName());
+        infoUtils.readLegacyDeploymentInfo(sgTwo);
+        infoUtils.checkMissingInOutputMemory(cliTestApp1War.getName());
+    }
+
+    /**
+     * Test undeploying operation by one application deployment to undeploy all by wildcard in all server group.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testUndeployWithAllServerGroups() throws Exception {
         ctx.handle("deployment deploy-file --server-groups=" + sgOne + ',' + sgTwo + " " + cliTestApp1War.getAbsolutePath());
 
-        checkState(sgOne, cliTestApp1War.getName(), true);
-        checkState(sgTwo, cliTestApp1War.getName(), true);
+        infoUtils.checkDeploymentByInfo(sgOne, cliTestApp1War.getName(), ENABLED);
+        infoUtils.checkDeploymentByInfo(sgTwo, cliTestApp1War.getName(), ENABLED);
 
         // From serverGroup1 only, still referenced from sg2. Must keep-content
         ctx.handle("deployment undeploy --server-groups=" + sgOne + ' ' + cliTestApp1War.getName());
 
-        checkState(sgOne, cliTestApp1War.getName(), false);
-        checkState(sgTwo, cliTestApp1War.getName(), true);
-    }
+        infoUtils.checkDeploymentByInfo(sgOne, cliTestApp1War.getName(), ADDED);
+        infoUtils.checkDeploymentByInfo(sgTwo, cliTestApp1War.getName(), ENABLED);
 
-    private void checkState(String groupName, String deploymentname, boolean expected) throws Exception {
```

Comment:
Again this way to check the state of a deployment is right and has proven itself to be valid. I advise that you integrate these checks in your tests.